lib/fizzy/parser.cpp

```diff
@@ -331,9 +331,17 @@ Module parse(bytes_view input)
     input.remove_prefix(wasm_prefix.size());
 
     Module module;
+    SectionId last_id = SectionId::custom;
     for (auto it = input.begin(); it != input.end();)
     {
         const auto id = static_cast<SectionId>(*it++);
+        if (id != SectionId::custom)
+        {
+            if (id <= last_id)
+                throw parser_error{"Unexpected out-of-order section type"};
```

Comment:
Because this is part of (noncritical) validation there should be option to disable this check. It should also throw `validation_error` (new exception type).

Reply:
Well it is part of the binary format section and not the validation section, hence I believe `parser_error` is more appropriate.

Reply:
Ok then.